drivers/sensor/i3g4250d/i3g4250d.c

```diff
@@ -0,0 +1,230 @@
+/* ST Microelectronics I3G4250D gyro driver
+ *
+ * Copyright (c) 2021 Jonathan Hahn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/i3g4250d.pdf
+ */
+
+#define DT_DRV_COMPAT st_i3g4250d
+
+#include <init.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#include "i3g4250d.h"
+
+#define RAW_TO_MICRODEGREEPERSEC 8750
+
+LOG_MODULE_REGISTER(i3g4250d, CONFIG_SENSOR_LOG_LEVEL);
+
+static int i3g4250d_sample_fetch(const struct device *dev,
+			enum sensor_channel chan)
+{
+
+	struct i3g4250d_data *i3g4250d = dev->data;
+	int ret;
+	uint8_t reg;
+	int16_t buf[3] = { 0 };
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_GYRO_XYZ);
+
+	ret = i3g4250d_flag_data_ready_get(i3g4250d->ctx, &reg);
```

Comment:
in case `int != int32_t` you may also add:

```suggestion
	ret = (int)i3g4250d_flag_data_ready_get(i3g4250d->ctx, &reg);
```

Reply:
Looked closer into it and it returns what the ctx functions return, so this is not needed and return is now a regular int.